client/src/app/modules/events/event-edit/event-edit.component.ts

```diff
@@ -10,22 +10,26 @@ import { IEventSchema } from '../../../interfaces';
 })
 export class EventEditComponent implements OnInit {
   event: IEventSchema;
-  isCreate = false;
+  isCreate: boolean = false;
 
   constructor(
     private eventService: EventService,
     private router: Router,
     private route: ActivatedRoute
   ) {
     this.event = new IEventSchema();
+    this.event.location = new IEventLocation();
    }
 
-  ngOnInit() {
+  ngOnInit(): void {
     if (this.route.snapshot.params['id'] === 'new') {
       this.isCreate = true;
     } else {
       this.eventService.getEvent(this.route.snapshot.params['id'])
         .subscribe((event) => {
+          if (!event.location) {
+            event.location = new IEventLocation();
```

Comment:
Not sure if it is still needed, it's not nice but don't remember why I had an mandatory property undefined